### DISCUSSION TITLE
Volume calculator results for roast properties are not reflected

### DISCUSSION THREAD

**Discussion by larue3** [artisan-roaster-scope/artisan#1511] [Q&A]
Thank you for the wonderful software. I have a question.

The volume value updates when using the volume calculator, but when I reopen the roast properties, the value disappears.

I tried several versions and it was recording successfully until v2.82. It seems that the value is not recorded in versions after v2.84.

What about you?
I'm using Windows 11, 23H2.
I had the same problem on Windows 10.

Thanks

**Comment by roasterdave** [comment]
Works fine here on Windows.  Can you provide a detailed step-by-step we can use to reproduce what you are seeing?

**Reply by larue3** [reply]
Hello roasterdave
Thank you for your reply.

-The first image is based on v2.8.2.
-The second image is based on v2.10.2.
-The third image is the auto-save prefix description and its file name.

For each roasting curve, we used samples created in the designer.

The difference between the two versions is that the "~roastgain" item does not seem to have been updated. The item appeared to be updated on the profile, but when I reopened the roast profile, the item was gone. It is not displayed on the graph either.

The item is also shown as 0% in the file name that was actually exported.

I have three PCs, all of which are old. 2011-2013
i5-2410M windows10
i5-3340M windows10
i7-3770 windows11
Both PCs seem to have the same symptoms.

![2024-03-13-103945](https://github.com/artisan-roaster-scope/artisan/assets/163118025/5f90a452-3645-42b1-8c93-7df90086ffd6)
![2024-03-13-104813](https://github.com/artisan-roaster-scope/artisan/assets/163118025/6b9fc77f-711e-42df-b24e-9ab9be5d03c6)
![2024-03-13-105250](https://github.com/artisan-roaster-scope/artisan/assets/163118025/68129f31-1e3d-41ec-b5e8-cd5db8839e3f)

**Reply by MAKOMO** [reply]
Thanks for this detailed description. Could you please test the current [continuous build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous) and report back if that performs any better?=

**Reply by larue3** [reply]
The "~volume" value in the volume calculation results has been updated correctly. (v2.10.3)

What I am concerned about is that the green and roasted values of ~weight may be displayed slightly different from the input values.
A 350 input was displayed as 350.02 and a 290 input was displayed as 290.01.

I'll try it for a while.
thank you.

![スクリーンショット-2024-03-14-081356](https://github.com/artisan-roaster-scope/artisan/assets/163118025/548acaea-5777-416b-a5bf-198906eaf681)
![スクリーンショット-2024-03-14-073105](https://github.com/artisan-roaster-scope/artisan/assets/163118025/12e8f312-1785-46fd-b2dc-d79e0a335638)

**Comment by larue3** [comment]
First image: v2.10.3 In the roast property, write 1500 for weight green and 1250 for roasted.

2nd image: File name where new data was exported using Save As.

As a result of repeated verification, it appeared that the ~weight value written in the roast property was confused somewhere.

thank you.

![スクリーンショット-2024-03-15-105438](https://github.com/artisan-roaster-scope/artisan/assets/163118025/b7745cd7-6e10-44a6-977c-5b41bda55308)

![スクリーンショット-2024-03-15-110224](https://github.com/artisan-roaster-scope/artisan/assets/163118025/5535d0a1-63ad-4c7c-8ace-1ea974ce83b7)

**Reply by MAKOMO** [reply]
I have no idea on the empty weight. Unlikely that the weight got lost within the app. I checked the code.

However, I revised the auto-calculation in the Roast Properties dialog again and your issues with the changing values should be resolved now. Could you download and test the latest continuous build again and please report back if this sthbilized now?

Thanks!

**Reply by larue3** [reply]
Thank you, Sir MAKOMO.
I salute you for your great work, your continued efforts, and your spirit of exploration.

This is my mistake.
I found out the reason why my filename was strange.

It seems that I used a character (_) that should not be used in the autosave prefix.

When I changed the prefix delimiter from "_" to "." (period) as shown below, the bean weight was recorded correctly.

~datetime_long.~roastingnotes_line.~cuppingnotes_line.~beans.~weight~weightunits.~weightlossp~volumegainp~aucc.~ambhumidityp~ambtempc

I apologize for wasting your precious time due to my mistake.
I wish you good luck.

Best regards

![スクリーンショット-2024-03-18-204827](https://github.com/artisan-roaster-scope/artisan/assets/163118025/a7c6802f-6765-4f6e-8427-3b79e274f6fb)

**Reply by larue3** [reply]
The automatic calculation results are also perfect.
Everything I was concerned about seems to have the correct values.
thank you
